bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/list/CommerceListImpl.java

```diff
@@ -0,0 +1,232 @@
+    static final String RESOURCE_TYPE = "core/cif/components/commerce/list/v1/list";
+    static final String LIST_SOURCE_PRODUCT_ASSOCIATION = "productAssociation";
+    static final String LIST_SOURCE_CATEGORY_ASSOCIATION = "categoryAssociation";
+    static final String LIST_SOURCE_STATIC = "static";
+    static final String PN_MAX_ITEMS = "maxItems";
+    static final String PN_PRODUCT = "product";
+    static final String PN_CATEGORY = "category";
+
+    @OSGiService
+    private ModelFactory modelFactory;
+
+    @Self
+    @Via(type = ForcedResourceType.class, value = CORE_WCM_LIST_RESOURCE_TYPE)
+    private List wcmList;
+
+    @ValueMapValue(name = List.PN_SOURCE, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String listFrom;
+
+    @ValueMapValue(name = PN_MAX_ITEMS, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private int maxItems;
+
+    @ValueMapValue(name = PN_PRODUCT, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String product;
+
+    @ValueMapValue(name = PN_CATEGORY, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String category;
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @Self
+    private SiteStructure siteStructure;
+
+    @OSGiService
+    private AssociatedContentService associatedContentService;
+
+    @OSGiService
+    private UrlProvider urlProvider;
+
+    @PostConstruct
+    void initModel() {
+        AssociatedContentQuery<Page> contentQuery = null;
+
+        if (LIST_SOURCE_PRODUCT_ASSOCIATION.equals(listFrom)) {
+            String sku = StringUtils.isNotBlank(product) ? product
+                : siteStructure.isProductPage(currentPage) ? urlProvider.getProductIdentifier(request) : null;
+
+            if (StringUtils.isNotBlank(sku)) {
+                contentQuery = associatedContentService.listProductContentPages(resourceResolver,
+                    PageParams.of(sku).path(siteStructure.getLandingPage().getPath()));
+            } else {
+                LOGGER.warn("Product SKU missing for Commerce List at " + resource.getPath());
+            }
+        } else if (LIST_SOURCE_CATEGORY_ASSOCIATION.equals(listFrom)) {
+            String categoryUid = StringUtils.isNotBlank(category) ? category
+                : siteStructure.isCategoryPage(currentPage) ? urlProvider.getCategoryIdentifier(request) : null;
+            if (StringUtils.isNotBlank(categoryUid)) {
+                contentQuery = associatedContentService.listCategoryContentPages(resourceResolver,
+                    PageParams.of(categoryUid).path(siteStructure.getLandingPage().getPath()));
+            } else {
+                LOGGER.warn("Category identifier missing for Commerce List at " + resource.getPath());
+            }
+        }
+
+        if (contentQuery == null) {
```

Comment:
Iiuc this causes the component to not support any of the WCM List's behaviours right? Shouldn't we implement a fallback here? Otherwise customers would end up with two list components, one for the wcm case and one for the cif case.

Ideally we could just create the super type model without providing any additional parameters in this case. This would make the Commerce List component a drop in replacement for the wcm list.

Reply:
Actually the wcmList is already initialized via injection, so nothing to do there. 
My intention is to provide a drop in replacement  for WCM List.